### PR TITLE
Kevin/connect bookmark likes

### PR DESCRIPTION
**Pull request by kevingreer**

### CODE REVIEW

File: Podcast/BookmarkViewController.swift [hunk 1]
```diff
@@ -49,6 +52,8 @@ class BookmarkViewController: UIViewController, UITableViewDelegate, UITableView
             }
         }
         bookmarkTableView.reloadData()
+        
+        fetchEpisodes()
```
Comment (natashaarmbrust) [inline Podcast/BookmarkViewController.swift]: you probably wanna fetch episodes then reload data

File: Podcast/BookmarkViewController.swift [hunk 2]
```diff
@@ -113,7 +119,15 @@ class BookmarkViewController: UIViewController, UITableViewDelegate, UITableView
     
     func bookmarkTableViewCellDidPressMoreActionsButton(bookmarksTableViewCell: BookmarkTableViewCell) {
         let option1 = ActionSheetOption(title: "Download", titleColor: .cancelButtonRed, image: #imageLiteral(resourceName: "more_icon"), action: nil)
-        let option2 = ActionSheetOption(title: "Delete Bookmark", titleColor: .podcastBlack, image: #imageLiteral(resourceName: "more_icon"), action: nil)
+        let option2 = ActionSheetOption(title: "Delete Bookmark", titleColor: .podcastBlack, image: #imageLiteral(resourceName: "more_icon")) {
+            let deleteBookmarkEndpointRequest = DeleteBookmarkEndpointRequest(episodeID: bookmarksTableViewCell.episodeID)
```
Comment (natashaarmbrust) [inline Podcast/BookmarkViewController.swift:135]: I think we should delete the episode in the deleteBookmarkEndpointRequest success closure

File: Podcast/BookmarkViewController.swift [hunk 3]
```diff
@@ -134,14 +148,14 @@ class BookmarkViewController: UIViewController, UITableViewDelegate, UITableView
...
-        let episode = Episode()
-        episode.title = "Puppies Galore"
-        episode.seriesTitle = "Amazing Doggos"
-        episode.dateCreated = Date()
-        episode.descriptionText = "We talk lots about dogs and puppies and how cute they are and the different colors they come in and how fun they are."
-        episode.tags = [Tag(name:"Design"), Tag(name:"Learning"), Tag(name: "User Experience"), Tag(name:"Technology"), Tag(name:"Innovation"), Tag(name:"Dogs")]
-        episode.numberOfRecommendations = 1482386868
-        
-        episodes = Array(repeating: episode, count: 5)
+        let endpointRequest = FetchBookmarksEndpointRequest()
```
Comment (natashaarmbrust) [inline Podcast/BookmarkViewController.swift]: rename to fetchBookmarksEndpointRequest

File: Podcast/CreateBookmarkEndpointRequest.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+//
+
+import Foundation
+import SwiftyJSON
+
+class CreateBookmarkEndpointRequest: EndpointRequest {
+    
+    var episodeID: String
+    
+    init(episodeID: String) {
```
Comment (natashaarmbrust) [inline Podcast/CreateBookmarkEndpointRequest.swift:16]: mad fucking spaces lol ... make this have less spaces

File: Podcast/CreateRecommendationEndpointRequest.swift [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+//
+
+import Foundation
+import SwiftyJSON
+
+class CreateRecommendationEndpointRequest: EndpointRequest {
+    
+    var episodeID: String
+    
+    init(episodeID: String) {
```
Comment (natashaarmbrust) [inline Podcast/CreateRecommendationEndpointRequest.swift:16]: same here with the spaces

File: Podcast/DeleteBookmarkEndpointRequest.swift [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+import Foundation
+import SwiftyJSON
+
+class DeleteBookmarkEndpointRequest: EndpointRequest {
+    
+    var episodeID: String
+    
+    init(episodeID: String) {
```
Comment (natashaarmbrust) [inline Podcast/DeleteBookmarkEndpointRequest.swift:16]: again spaces

File: Podcast/DeleteRecommendationEndpointRequest.swift [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+//  Created by Kevin Greer on 5/2/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+
+class DeleteRecommendationEndpointRequest: EndpointRequest {
+    
+    var episodeID: String
```
Comment (natashaarmbrust) [inline Podcast/DeleteRecommendationEndpointRequest.swift:14]: reduce spaces

File: Podcast/Episode.swift [hunk 8]
```diff
@@ -56,13 +56,13 @@ class Episode: NSObject {
     }
     
      convenience init(json: JSON) {
-        let id = json["id"].stringValue
+        let id = json["id"].string ?? json["episodeId"].stringValue
```
Comment (natashaarmbrust) [inline Podcast/Episode.swift:59]: does backend return episodeId sometimes and id othertimes?

File: Podcast/Episode.swift [hunk 9]
```diff
@@ -118,5 +118,25 @@ class Episode: NSObject {
         
         return attrStr
     }
+    
+    func saveRecommendedState() {
+        if isRecommended {
+            let endpointRequest = CreateRecommendationEndpointRequest(episodeID: id)
```
Comment (natashaarmbrust) [inline Podcast/Episode.swift]: change name to createRecommendedEndpointRequest

Comment (natashaarmbrust) [reply Podcast/Episode.swift]: on success / failure you should be changing the state

Comment (natashaarmbrust) [inline Podcast/Episode.swift]: also not comfortable with this being in the model

File: Podcast/Episode.swift [hunk 10]
```diff
@@ -118,5 +118,25 @@ class Episode: NSObject {
...
+            System.endpointRequestQueue.addOperation(endpointRequest)
+        } else {
+            let endpointRequest = DeleteRecommendationEndpointRequest(episodeID: id)
+            System.endpointRequestQueue.addOperation(endpointRequest)
+        }
+    }
+    
+    func saveBookmarkedState() {
+        if isBookmarked {
+            let endpointRequest = CreateBookmarkEndpointRequest(episodeID: id)
```
Comment (natashaarmbrust) [inline Podcast/Episode.swift]: same comments as above

File: Podcast/FetchBookmarksEndpointRequest.swift [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+
+class FetchBookmarksEndpointRequest: EndpointRequest {
+    
+    override init() {
+        super.init()
```
Comment (natashaarmbrust) [inline Podcast/FetchBookmarksEndpointRequest.swift:15]: remove spaces

File: Podfile.lock [hunk 12]
```diff
@@ -80,4 +80,4 @@ SPEC CHECKSUMS:
```
Comment (natashaarmbrust) [inline Podfile.lock]: uncommit this file

File: Podcast/TagViewController.swift [hunk 13]
```diff
@@ -163,11 +163,13 @@ class TagViewController: UIViewController, UITableViewDelegate, UITableViewDataS
     func recommendedEpisodeOuterTableViewCellDidPressBookmarkButton(episodeTableViewCell: EpisodeTableViewCell, episode: Episode) {
         episode.isBookmarked = !episode.isBookmarked
         episodeTableViewCell.setBookmarkButtonToState(isBookmarked: episode.isBookmarked)
+        episode.saveBookmarkedState()
     }
```
Comment (natashaarmbrust) [inline Podcast/TagViewController.swift]: we discussed handling this in controllers

**Comment by natashaarmbrust** [comment]
after making all these changes go ahead and merge

File: Podcast/Episode.swift (hunk 8 above)
Comment (kevingreer) [reply Podcast/Episode.swift:59]: yeah, joe could change it or we could keep this

File: Podcast/BookmarkViewController.swift (hunk 2 above)
Comment (kevingreer) [reply Podcast/BookmarkViewController.swift:135]: isn't that what I'm doing?

**Pull request merged**